scripts/dupe_finder.py

```diff
@@ -0,0 +1,72 @@
+import json
+import os
+import hashlib
+
+
+def get_file_hash(filename, alg, chunksize=131072):
+    if (alg == 'sha256'):
+        h = hashlib.sha256()
+    elif (alg == 'sha1'):
+        h = hashlib.sha1()
+    elif (alg == 'md5'):
+        h = hashlib.md5()
+
+    with open(filename, 'rb', buffering=0) as f:
+        for b in iter(lambda: f.read(chunksize), b''):
+            h.update(b)
+    return h.hexdigest()
+
+
+'''
+In JSON we have multiple files for each rom. This could lead sometimes to not have any executable file in that list.
+For each file in "files" field in JSON, take the extension.
+If that file is a homebrew, returns its path, otherwise, if files specified in the list of files in the JSON does not contain any executable gb file, return -1.
+'''
+def look_for_rom(files):
+    for f in range(0, len(files)):
+        ext = files[f]['filename'].split('.')[-1]
+        if ext.lower() in ['gb', 'gbc', 'cgb', 'gba', 'agb', 'sgb']:
+            return files[f]['filename']
+
+    return -1
+
+
+d = dict()      # a dictionary has been created since dictionaries are really useful to detect unique things.
```

Comment:
This comment sounds just confusing ... is it okay to leave it undocumented if in the following logic you explain how you are gonna use it, but to say "we do X because X is useful" doesn't seem to add anything :/

Reply:
my bad, lazy copy-pasting bring me to this